lib/anyway/config.rb

```diff
@@ -126,14 +128,36 @@ def config_attributes
         end
       end
 
-      def required(*names)
-        unless (unknown_names = (names - config_attributes)).empty?
-          raise ArgumentError, "Unknown config param: #{unknown_names.join(",")}"
-        end
+      def required(*names, env: nil)
```

Comment:
@fargelus I like the idea of filtering by env at load time, not during the config instance initialization, proposed by @deflexor [here](https://github.com/palkan/anyway_config/pull/98/files#diff-046f0f464527576496ebd252f1ff8eee64433d0d8caa71b4a09a7d5cac4f3c93R129-R154).

I think, it's safe to assume (at least, for Rails apps) that the environment is defined prior to loading any Anyway Config class (for non-Rails apps we should add a note about this to the docs).

This assumption could help us to drastically simplify the code.

Reply:
🤔